monkey/agent_plugins/exploiters/smb/src/plugin.py

```diff
@@ -17,13 +17,19 @@
     generate_brute_force_credentials,
 )
 from infection_monkey.exploit.tools.helpers import get_agent_dst_path
-from infection_monkey.i_puppet import ExploiterResultData, TargetHost
+from infection_monkey.i_puppet import ExploiterResultData, PortScanData, TargetHost
 from infection_monkey.propagation_credentials_repository import IPropagationCredentialsRepository
 
 from .smb_command_builder import build_smb_command
 from .smb_options import SMBOptions
+from .smb_remote_access_client import SMB_PORTS
 from .smb_remote_access_client_factory import SMBRemoteAccessClientFactory
 
+
+def is_open_smb_port(psd: PortScanData) -> bool:
+    return psd.status == PortStatus.OPEN and psd.port in SMB_PORTS
```

Comment:
The logic here is incomplete.

If all SMB ports are closed, don't run
If any SMB ports are open, run
If the status of any SMB ports is unknown, run

Reply:
How are we modeling an unknown status? Or do you mean unknown service?

Reply:
If the port isn't listed in the TargetHost at all, its status is unknown (i.e. neither open nor closed.)